src/librustc_resolve/lib.rs

```diff
@@ -202,6 +202,12 @@ pub enum ResolutionError<'a> {
     AttemptToUseNonConstantValueInConstant,
 }
 
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum UnresolvedNameContext {
```

Comment:
Could you add a doc comment on this so it's not confusing for future hackers?

(Also explain what the nodeid is)